demos/twa-play-billing/src/main/java/com/google/androidbrowserhelper/demos/playbilling/AcknowledgeCall.java

```diff
@@ -0,0 +1,55 @@
+package com.google.androidbrowserhelper.demos.playbilling;
+
+import android.os.Bundle;
+import android.os.RemoteException;
+
+import androidx.annotation.Nullable;
+import androidx.browser.trusted.TrustedWebActivityCallbackRemote;
+
+public class AcknowledgeCall {
+    public static final String COMMAND_NAME = "acknowledge";
+
+    private static final String PARAM_ACKNOWLEDGE_PURCHASE_TOKEN = "acknowledge.purchaseToken";
+    private static final String PARAM_ACKNOWLEDGE_MAKE_AVAILABLE_AGAIN =
+            "acknowledge.makeAvailableAgain";
+    private static final String RESPONSE_ACKNOWLEDGE = "acknowledge.response";
+    private static final String RESPONSE_ACKNOWLEDGE_RESPONSE_CODE = "acknowledge.responseCode";
+
+    public final String purchaseToken;
+    public final boolean makeAvailableAgain;
+    private final TrustedWebActivityCallbackRemote mCallback;
+
+    private AcknowledgeCall(String purchaseToken, boolean makeAvailableAgain,
```

Comment:
I am missing some documentation on why this needs to be private and created through create(...) instead of a public constructor. It seems like the class is not a singleton.

Reply:
So I don't really have too strong of a reason why this constructor is private, other than it doesn't need to be public at the moment. But I do have a reason why I made the static `create` method (instead of having that logic in a constructor).

We're creating the object from the `args` bundle, which may or may not be valid. If we had a constructor that was given an invalid `args` (for example, if it doesn't contain a purchase token) we could either:
- Construct an `AcknowledgeCall` that is incomplete (eg, `purchaseToken` is null).
- Throw an exception.

On the other hand, we use a static `create` method, we have another option - we can return `null`.

It's better if `AcknowledgeCall` can't represent invalid values - either "it exists and its valid" or "it doesn't exist". The difference between the other two is mostly ergonomics - would we prefer the caller to have to use a try-catch or an `if (thing == null)`, but also a bit of performance (Java exceptions have some overhead).

So I would definitely prefer to have a `create(Bundle args)` over a `new AcknowledgeCall(Bundle args)`. However, if there are other things that construct this object, I would be fine also having a public `new Acknowledge(String purchaseToken, boolean makeAvailableAgain)`.